internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -49,7 +49,9 @@ func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AServ
 		return nil, err
 	}
 
-	// TODO(rmehta19): Handle resp.GetStatus().
+	if resp.GetStatus().Code != 0 {
+		return nil, errors.New(resp.GetStatus().Details)
+	}
```

Comment:
While we are at it, let's also add a unit test to cover this if branch. This would correspond to any internal error returned by the S2A service to the S2A client.

Reply:
Will do! I added a test for this branch in the fakes2av2_test suite